### PR TITLE
Chainlink historical price

### PR DESCRIPTION
**Pull request by haythem96**
# Task: Chainlink historical price

## High Level Description

Should be reviewed/merged after #369 .

This PR add the functionality to get historical prices from chainlink feed. If the asset is a stable asset, the Oracle will return always the same price with the block timestamp as round timestamp.

Changes made:
- Re-ordered functions
- Added `getChainlinkRoundData()` in `Oracle.sol`
- Added `getHistoricalPrice()` in `ChainlinkPricer.sol`
- Removed the inheritance of `OpynPricerInterface` from `CompoundPricer` as that pricer do not have a historical price function with a round ID.

### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [x] Is your code up to date with the spec? 
- [x] Have you added your tests to the testing doc?

### CODE REVIEW

File: contracts/Oracle.sol [hunk 1]
```diff
@@ -63,6 +63,119 @@ contract Oracle is Ownable {
...
+     * @param _price asset price at expiry
+     */
+    function setExpiryPrice(
+        address _asset,
+        uint256 _expiryTimestamp,
+        uint256 _price
+    ) external {
+        require(msg.sender == assetPricer[_asset], "Oracle: caller is not authorized to set expiry price");
+        require(isLockingPeriodOver(_asset, _expiryTimestamp), "Oracle: locking period is not over yet");
+        require(storedPrice[_asset][_expiryTimestamp].timestamp == 0, "Oracle: dispute period started");
```
Comment (antoncoding) [inline contracts/Oracle.sol:173]: can we change this error message to `Oracle: price already submitted`? I think mentioning dispute period is not making it easier to understand

File: contracts/Oracle.sol [hunk 2]
```diff
@@ -139,6 +252,26 @@ contract Oracle is Ownable {
         return pricerDisputePeriod[_pricer];
     }
 
+    /**
+     * @notice get historical asset price
```
Comment (antoncoding) [inline contracts/Oracle.sol:256]:
```suggestion
     * @notice get historical asset price and timestamp
```

File: contracts/Oracle.sol (hunk 1 above)
Comment (haythem96) [reply contracts/Oracle.sol:173]: Will do in another PR

**Comment by antoncoding** [comment]
can we change the base branch to #369 in these cases? Otherwise it's hard to review (don't know what's new in this vs new in its parent) and leave comments at the right place.

**Review by antoncoding — APPROVED** [review]

**Comment by aparnakr** [comment]
> can we change the base branch to #369 in these cases? Otherwise it's hard to review (don't know what's new in this vs new in its parent) and leave comments at the right place.

I can still see all the changes from that branch as well :( Any way to merge with that?

**Comment by haythem96** [comment]
> > can we change the base branch to #369 in these cases? Otherwise it's hard to review (don't know what's new in this vs new in its parent) and leave comments at the right place.
> 
> I can still see all the changes from that branch as well :( Any way to merge with that?

No it is showing only new changes in this PR now!

**Review by aparnakr — APPROVED** [review]

**Pull request merged**